### PR TITLE
chore: Cleanup before release

### PR DESCRIPTION
**Pull request by manusa**
## Description

chore: Cleanup before release

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,8 +6,9 @@
 * Fix #2596: Add buildable references for Container, Port and Volume
 
 #### Improvements
-* Eliminated the use of Doneables and simplified the internal DSL implementation.
-* Provide a `customResources` alternative that simply accepts the resource type as an argument.
+* Fix #2571: Eliminated the use of Doneables and simplified the internal DSL implementation.
+* Fix #2607: add isMarkedForDeletion and finalizer-related default methods to HasMetadata (@metacosm)
+* Fix #2605: Provide a `customResources` alternative that simply accepts the resource type as an argument.
```
Comment (rohanKanojia) [inline CHANGELOG.md:11]: Shall we add a breaking changes section too?

**Review by rohanKanojia — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (manusa) [reply CHANGELOG.md:11]: Well, almost everything is going to break, not sure if it applies in this case.

We're also releasing an alpha version, API is bound to change after people start testing this version.

**Comment by manusa** [comment]
[merge]

**Comment by manusa** [comment]
[merge]

**Comment by manusa** [comment]
[merge]

**Comment by manusa** [comment]
[merge] Please!

**Comment by rohanKanojia** [comment]
Looks like Jenkins is sick today :laughing: . You can try merging directly from Github UI. I don't think this would cause any problems.

**Comment by manusa** [comment]
It seems that the local .m2 repo for the node is somewhat corrupted.
Could you change mvn build command and add a -U flag?

**Comment by manusa** [comment]
Anyway, I'm manually merging so I can start the release process

**Pull request merged**